infrastructure/tasks/20-apim_setup.ts

```diff
@@ -120,30 +133,39 @@ const setupConfigurationFromGit = async (
     }
   );
 
+  // Build Git URL with user and password
   const { hostname, protocol } = url.parse(scmUrl);
   const scmUrlWithCreds = url.format({
     ...{ hostname, protocol },
     auth: `${config.apim_scm_username}:${gitKey.value}`
   });
 
   // Push master branch
+  // Actually, the only things we need to push are the developer portal templates
+  // as there is no API to edit them programmatically
   const tmpDir = tmp.dirSync();
   if (!tmpDir) {
     throw new Error("Cannot create temporary directory");
   }
   shelljs.cp("-R", configurationDirectoryPath, tmpDir.name);
   shelljs.pushd(path.join(tmpDir.name, CONFIGURATION_DIRECTORY_NAME));
+
+  const changes = replaceVariables(tmpDir.name, config);
+  console.log(
+    `Replaced configuration parameters in ${JSON.stringify(changes)}`
+  );
+
   shelljs.exec(`git init .`);
```

Comment:
this code is not idempotent, what happens if something breaks in between of an execution?

Reply:
can you please elaborate more ?

Reply:
for instance, if `shelljs.exec(`git push origin master --force`);` fails, you have an inconsistent state

if you run this task again, it will attempt to run `shelljs.exec(`git init .`);` and fail (since that's already a git repo) ?

Reply:
nope because it acts on a copy of the whole repo to a temporary directory